toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -223,74 +234,119 @@ func (b *LiveOSIsoBuilder) prepareRootfsForDracut(writeableRootfsDir string) err
-	}
+	grubMkconfigEnabled := isGrubMkconfigConfig(inputContentString)
+	if !grubMkconfigEnabled {
+		var oldLinuxPath string
+		inputContentString, oldLinuxPath, err = setLinuxPath(inputContentString, isoKernelPath)
+		if err != nil {
+			return fmt.Errorf("failed to update the kernel file path in the iso grub.cfg:\n%w", err)
+		}
 
-	inputContentString, err = replaceToken(inputContentString, oldInitrdPath, isoInitrdPath)
-	if err != nil {
-		return fmt.Errorf("failed to update all the initrd file path occurances in the iso grub.cfg:\n%w", err)
+		inputContentString, err = replaceToken(inputContentString, oldLinuxPath, isoKernelPath)
+		if err != nil {
+			return fmt.Errorf("failed to update all the kernel file path occurances in the iso grub.cfg:\n%w", err)
+		}
+
+		var oldInitrdPath string
+		inputContentString, oldInitrdPath, err = setInitrdPath(inputContentString, isoInitrdPath)
+		if err != nil {
+			return fmt.Errorf("failed to update the initrd file path in the iso grub.cfg:\n%w", err)
+		}
+
+		inputContentString, err = replaceToken(inputContentString, oldInitrdPath, isoInitrdPath)
+		if err != nil {
+			return fmt.Errorf("failed to update all the initrd file path occurances in the iso grub.cfg:\n%w", err)
+		}
+	} else {
+		inputContentString, _, err = setLinuxOrInitrdPathAll(inputContentString, linuxCommand, isoKernelPath, true /*allowMultiple*/)
+		if err != nil {
+			return fmt.Errorf("failed to update the kernel file path in the iso grub.cfg:\n%w", err)
+		}
+
+		inputContentString, _, err = setLinuxOrInitrdPathAll(inputContentString, initrdCommand, isoInitrdPath, true /*allowMultiple*/)
+		if err != nil {
+			return fmt.Errorf("failed to update the initrd file path in the iso grub.cfg:\n%w", err)
+		}
 	}
 
-	inputContentString, _, err = replaceKernelCommandLineArgValue(inputContentString, "root", rootValue)
+	rootValue := fmt.Sprintf(rootValueTemplate, isomakerlib.DefaultVolumeId)
+	inputContentString, _, err = replaceKernelCommandLineArgValueAll(inputContentString, "root", rootValue, true /*allowMultiple*/)
 	if err != nil {
 		return fmt.Errorf("failed to update the root kernel argument in the iso grub.cfg:\n%w", err)
 	}
 
-	inputContentString, err = updateSELinuxCommandLineHelper(inputContentString, imagecustomizerapi.SELinuxModeDisabled)
+	inputContentString, err = updateSELinuxCommandLineHelperAll(inputContentString, imagecustomizerapi.SELinuxModeDisabled)
 	if err != nil {
 		return fmt.Errorf("failed to set SELinux mode:\n%w", err)
 	}
 
-	liveosKernelArgs := fmt.Sprintf(kernelArgsTemplate, liveOSDir, liveOSImage, extraCommandLine)
+	mergedExtraCommandLine, err := mergeKernelParameters(savedConfigKernelArgsFilePath, extraCommandLine)
+	if err != nil {
+		return fmt.Errorf("failed to combine additional kernel command line parameters:\n%w", err)
+	}
+
+	liveosKernelArgs := fmt.Sprintf(kernelArgsLiveOSTemplate, liveOSDir, liveOSImage)
+	additionalKernelCommandline := " " + liveosKernelArgs + " " + mergedExtraCommandLine
```

Comment:
`appendKernelCommandLineArgsAll` should handle adding spaces around the args.

Reply:
pending

Reply:
fixed.